examples/pytorch/torchscript/mnist_torchscript.py

```diff
@@ -0,0 +1,250 @@
+        )
+
+    def val_dataloader(self):
+        """
+        :return: output - Validation data loader for the given input
+        """
+        mnist_train = datasets.MNIST("dataset", download=True, train=True, transform=self.transform)
+        mnist_train, mnist_val = random_split(mnist_train, [55000, 5000])
+
+        return DataLoader(
+            mnist_val, batch_size=self.args["batch_size"], num_workers=self.args["num_workers"],
+        )
+
+    def test_dataloader(self):
+        """
+        :return: output - Test data loader for the given input
+        """
+        mnist_test = datasets.MNIST("dataset", download=True, train=False, transform=self.transform)
+        return DataLoader(
+            mnist_test, batch_size=self.args["batch_size"], num_workers=self.args["num_workers"],
+        )
+
+    def configure_optimizers(self):
+        """
+        Initializes the optimizer and learning rate scheduler
+        :return: output - Initialized optimizer and scheduler
+        """
+        self.optimizer = torch.optim.Adam(self.parameters(), lr=self.args["lr"])
+        self.scheduler = {
+            "scheduler": torch.optim.lr_scheduler.ReduceLROnPlateau(
+                self.optimizer, mode="min", factor=0.2, patience=2, min_lr=1e-6, verbose=True,
+            )
+        }
+        return [self.optimizer], [self.scheduler]
+
+    def optimizer_step(
+        self,
+        epoch,
+        batch_idx,
+        optimizer,
+        optimizer_idx,
+        second_order_closure=None,
+        on_tpu=False,
+        using_lbfgs=False,
+        using_native_amp=False,
+    ):
+        """
+        Training step function which runs for the given number of epochs
+        :param epoch: Number of epochs to train
+        :param batch_idx: batch indices
+        :param optimizer: Optimizer to be used in training step
+        """
+        self.optimizer.step()
+        self.optimizer.zero_grad()
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser(description="TorchScripted model example")
+
+    # Add trainer specific arguments
+    parser.add_argument(
+        "--tracking-uri", type=str, default="http://localhost:5000/", help="mlflow tracking uri"
+    )
+    parser.add_argument(
+        "--max-epochs", type=int, default=5, help="number of epochs to run (default: 5)"
+    )
+    parser.add_argument(
+        "--gpus", type=int, default=0, help="Number of gpus - by default runs on CPU"
+    )
+    parser.add_argument(
+        "--distributed-backend",
+        type=str,
+        default=None,
+        help="Distributed Backend - (default: None)",
+    )
+    parser = LightningMNISTClassifier.add_model_specific_args(parent_parser=parser)
+
+    args = parser.parse_args()
+    dict_args = vars(args)
+    mlflow.set_tracking_uri(dict_args["tracking_uri"])
+    model = LightningMNISTClassifier(**dict_args)
+    scripted_model = torch.jit.script(model)
```

Comment:
Shouldn't this scripted model be saved after the model has been trained? The sequence of steps in the example is not clear